src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -765,10 +767,21 @@ def clone_subvolume_snapshot(self, **kwargs):
         s_groupname  = kwargs['group_name']
 
         try:
-            with open_volume(self, volname) as fs_handle:
-                with open_group(fs_handle, self.volspec, s_groupname) as s_group:
-                    with open_subvol(self.mgr, fs_handle, self.volspec, s_group, s_subvolname, SubvolumeOpType.CLONE_SOURCE) as s_subvolume:
-                        self._clone_subvolume_snapshot(fs_handle, volname, s_group, s_subvolume, **kwargs)
+            if get_all_pending_clones_count(self, self.mgr, self.volspec) >= self.mgr.max_concurrent_clones:
+                ve = VolumeException(-errno.EAGAIN, f"clone creation request exceeds maximum limit")
+                ret = self.volume_exception_to_retval(ve)
+                return ret
+            try:
```

Comment:
The code is functionally correct. Could you re-org this for better readability?
1. Let's not have two try except block. One should suffice.
2. Let's handle only `VolumeException` here ? The `get_all_pending_clones_count` can convert `IndexException` to `VolumeException`  ?

This way it looks much simpler to read.

Reply:
Yup, updated